contracts/0.8.9/SepoliaDepositAdapter.sol

```diff
@@ -4,24 +4,82 @@
 /* See contracts/COMPILERS.md */
 pragma solidity 0.8.9;
 
+import "@openzeppelin/contracts-v4.4/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts-v4.4/access/Ownable.sol";
 
-contract SepoliaDepositAdapter {
+import "hardhat/console.sol";
 
-    uint public constant TEST_VALUE = 16;
-    address public immutable depositContract;
+// Sepolia deposit contract variant of the source code https://github.com/protolambda/testnet-dep-contract/blob/master/deposit_contract.sol
+interface ISepoliaDepositContract is IERC20 {
 
     function deposit(
         bytes calldata pubkey,
         bytes calldata withdrawal_credentials,
         bytes calldata signature,
         bytes32 deposit_data_root
-    ) external payable {
-    }
+    ) external payable;
+
+    function get_deposit_root() external view returns (bytes32);
+
+    function get_deposit_count() external view returns (bytes memory);
+
+    function name() external view returns (string memory);
+}
+
+contract SepoliaDepositAdapter is Ownable {
+
+    uint public constant VERSION = 2;
+    ISepoliaDepositContract public immutable originalContract;
 
     constructor(address _deposit_contract) {
-        depositContract = _deposit_contract;
+        originalContract = ISepoliaDepositContract(_deposit_contract);
+    }
+
+    function get_deposit_root() external view returns (bytes32) {
+        return originalContract.get_deposit_root();
+    }
+
+    function get_deposit_count() external view returns (bytes memory) {
+        return originalContract.get_deposit_count();
     }
 
+    function test() external view returns (string memory) {
+        return originalContract.name();
+    }
+
+    receive() external payable {
+        console.log(
+          "Receive %s from %s",
+            msg.value,
+            msg.sender
+        );
+    }
 
+    function drain() external onlyOwner {
+        uint ownTokens = address(this).balance;
+        address payable _owner = payable(owner());
+        _owner.transfer(ownTokens);
```

Comment:
I recalled why transfer() is generally not recommended. 

https://www.rareskills.io/post/smart-contract-security#:~:text=Transfer()%20and%20send()%20can%20break%20with%20multi%2Dsignature%20wallets

Reply:
ok. I can do destination address as a parameter. Do you have other suggestions?